crypto/ess/ess_lib.c

```diff
@@ -289,73 +289,93 @@ static int ess_issuer_serial_cmp(const ESS_ISSUER_SERIAL *is, const X509 *cert)
     return ASN1_INTEGER_cmp(is->serial, X509_get0_serialNumber(cert));
 }
 
-/* Returns < 0 if certificate is not found, certificate index otherwise. */
-int ossl_ess_find_cert(const STACK_OF(ESS_CERT_ID) *cert_ids, X509 *cert)
+/*
+ * Find cert referenced by |cid| (if not NULL, else |cidv2|) in |certs|.
+ * If the cid{,v2} index is 0, the cert must be in the first in |certs| list.
+ * Return -1 on cid not found, -2 on other error, else position in |certs| list.
```

Comment:
You do not seem to be distinguishing the negative return values when calling it. So it might make sense to not bother here either. If at all I'd probably do it the other way around - i.e. -1 as a general error return and -2 for not found as a special value.

Reply:
Right, the distinction was so far not really used.
Yet after exporting `OSSL_ESS_check_signing_certs()` (see also below)
it is used to distinguish not found (now indicated by return value 0,
and (fatal/internal) error, now indicated by return value -1.

Reply:
I can only repeat that -1 for not found and -2 for general error is a strange pattern that is IMO not appearing anywhere else.

Reply:
Ah, improving this this would also have been part of the more bulky fixup that I did yesterday.
I've just taken over that part here again.